### PR TITLE
Update structure of HTTP BBEs

### PR DESCRIPTION
**Pull request by ayeshLK**
### Updated BBEs:
* HTTP Cient Endpoint
* HTTP Redirect
* HTTP Streaming
* HTTP Passthrough
* HTTP 1.1 to HTTP 2.0 Protocol Switch

### CODE REVIEW

File: examples/http-client-endpoint/http_client_endpoint.bal [hunk 1]
```diff
@@ -4,48 +4,28 @@ import ballerina/io;
 // Creates a new client with the backend URL.
 final http:Client clientEndpoint = check new ("http://postman-echo.com");
 
-public function main() {
+public function main() returns error? {
     // Sends a `GET` request to the specified endpoint.
     io:println("GET request:");
-    http:Response|error response = clientEndpoint->get("/get?test=123");
-    handleResponse(response);
+    json echoResponse = check clientEndpoint->get("/get?test=123");
```
Comment (shafreenAnfar) [inline examples/http-client-endpoint/http_client_endpoint.bal:10]: Lets make `echoResponse` to `resp`

File: examples/http-client-endpoint/http_client_endpoint.bal [hunk 2]
```diff
@@ -4,48 +4,28 @@ import ballerina/io;
...
+    io:println(postResponse.toJsonString());
 
     // Uses the `execute()` remote function for custom HTTP verbs.
     io:println("\nUse custom HTTP verbs:");
-    response = clientEndpoint->execute("COPY", "/get", "CUSTOM: Hello World");
+    http:Response response = check clientEndpoint->execute("COPY", "/get", "CUSTOM: Hello World");
 
     // The `get()`, `head()`, and `options()` have the optional headers parameter to send out headers,
-    response = clientEndpoint->get("/get",
+    response = check clientEndpoint->get("/get",
```
Comment (shafreenAnfar) [inline examples/http-client-endpoint/http_client_endpoint.bal:23]: Lets move it right the below the first `get` example and bind it to json

File: examples/http-client-endpoint/http_client_endpoint.bal [hunk 3]
```diff
@@ -4,48 +4,28 @@ import ballerina/io;
...
-        } else {
-            io:println("Invalid payload received:", msg.message());
-        }
-    } else {
-        io:println("Error when calling the backend: ",
-                            response.message());
-    }
+    // [Get the content type](https://docs.central.ballerina.io/ballerina/http/latest/classes/Response#getContentType) from the response.
+    string contentType = response.getContentType();
+    io:println("Content-Type: " + contentType);
```
Comment (shafreenAnfar) [inline examples/http-client-endpoint/http_client_endpoint.bal:28]: Can't we do this 

`io:println("Content-Type: " + resp.getContentType());`

File: examples/http-passthrough/http_passthrough.bal [hunk 4]
```diff
@@ -1,5 +1,4 @@
 import ballerina/http;
-import ballerina/log;
 
 http:Client clientEP = check new ("http://localhost:9092/hello");
```
Comment (shafreenAnfar) [inline examples/http-passthrough/http_passthrough.bal:3]: Lets use postman endpoint for this.

File: examples/http-passthrough/http_passthrough.bal [hunk 5]
```diff
@@ -8,25 +7,22 @@ service /passthrough on new http:Listener(9090) {
...
         // used to invoke the passthrough resource. The `forward()` function returns the response from the backend if
         // there are no errors.
-        return clientEP->forward("/", req);
+        string response = check clientEP->forward("/", req);
+        return response;
     }
 }
 
 // Sample hello world service.
 service /hello on new http:Listener(9092) {
```
Comment (shafreenAnfar) [inline examples/http-passthrough/http_passthrough.bal:21]: Remove this

File: examples/http-1-1-to-2-0-protocol-switch/http_1_1_to_2_0_protocol_switch.bal [hunk 6]
```diff
@@ -4,26 +4,16 @@ import ballerina/http;
 http:Client http2serviceClientEP =
         check new ("http://localhost:7090", {httpVersion: "2.0"});
 
-service /http11Service on new http:Listener(9090) {
+service / on new http:Listener(9090) {
```
Comment (ThisaruGuruge) [inline examples/http-1-1-to-2-0-protocol-switch/http_1_1_to_2_0_protocol_switch.bal:7]: Do we need this `/` here?

```suggestion
service on new http:Listener(9090) {
```

Comment (chamil321) [reply examples/http-1-1-to-2-0-protocol-switch/http_1_1_to_2_0_protocol_switch.bal:7]: @ThisaruGuruge `/` is not needed, but kept it intentionally to show the basePath

File: examples/http-client-endpoint/http_client_endpoint.bal [hunk 7]
```diff
@@ -4,48 +4,27 @@ import ballerina/io;
...
-        if (msg is json) {
-            io:println(msg.toJsonString());
-        } else {
-            io:println("Invalid payload received:", msg.message());
-        }
-    } else {
-        io:println("Error when calling the backend: ",
-                            response.message());
-    }
+    // [Get the content type](https://docs.central.ballerina.io/ballerina/http/latest/classes/Response#getContentType) from the response.
```
Comment (chamil321) [inline examples/http-client-endpoint/http_client_endpoint.bal:28]: Invalid doc comment

File: examples/http-passthrough/http_passthrough.bal [hunk 8]
```diff
@@ -1,32 +1,18 @@
...
-import ballerina/log;
 
-http:Client clientEP = check new ("http://localhost:9092/hello");
+http:Client clientEP = check new ("http://postman-echo.com");
 
-service /passthrough on new http:Listener(9090) {
+service / on new http:Listener(9090) {
 
     // The passthrough resource allows all HTTP methods since the resource configuration does not explicitly specify
     // which HTTP methods are allowed.
```
Comment (chamil321) [inline examples/http-passthrough/http_passthrough.bal:8]:
```suggestion
    // The passthrough resource allows all HTTP methods as the accessor is `default`.
```

File: examples/http-client-endpoint/http_client_endpoint.bal (hunk 7 above)
Comment (chamil321) [reply examples/http-client-endpoint/http_client_endpoint.bal:28]:
```suggestion
```

File: examples/http-passthrough/http_passthrough.bal [hunk 9]
```diff
@@ -1,32 +1,18 @@
...
-    resource function 'default .(http:Request req)
-            returns http:Response|error? {
+    resource function 'default passthrough(http:Request req)
+            returns json|error? {
```
Comment (chamil321) [inline examples/http-passthrough/http_passthrough.bal:10]: Lets not do the data binding here. Let's return the response itself

File: examples/http-passthrough/http_passthrough.client.out [hunk 10]
```diff
@@ -1,7 +1,3 @@
 # To invoke the service, use the cURL commands below.
-curl http://localhost:9090/passthrough -X POST
-Hello World!
 curl http://localhost:9090/passthrough -X GET
```
Comment (chamil321) [inline examples/http-passthrough/http_passthrough.client.out:2]:
```suggestion
curl http://localhost:9090/passthrough
```

**Comment by chamil321** [comment]
lengthValidation has failed
```
    [/examples/http-client-endpoint/http_client_endpoint.bal] Line number 26.
    [/examples/http-response-with-multiparts/http_response_with_multiparts.bal] Line number 45.
```

**Review by chamil321 — APPROVED** [review]

**Pull request merged**